repository/activity_repository/activity_repository.go

```diff
@@ -0,0 +1,40 @@
+package activity_repository
+
+import (
+	"encoding/json"
+	"gin/config"
+	"gin/models"
+	"gopkg.in/redis.v5"
+)
+
+func (rs RedisActivityRepository) GetUserActivity(username string) ([]string, error) {
+	return rs.client.LRange(username, 0, 2).Result()
+}
+
+func (rs RedisActivityRepository) SetUserActivity(username string, request models.UserRequest) error {
+	reqJson, err := json.Marshal(request)
+	if err != nil {
+		return err
+	}
+
+	cmd := rs.client.LPush(username, reqJson)
+	if err := cmd.Err(); err != nil {
+		return err
+	}
+
+	err = rs.client.LTrim(username, 0, 2).Err()
+	return err
+}
+
+type ActivityService interface {
+	GetUserActivity(username string) ([]string, error)
+	SetUserActivity(username string, request models.UserRequest) error
+}
+
+type RedisActivityRepository struct {
+	client *redis.Client
+}
+
+func GetActivityRepository() ActivityService {
```

Comment:
```suggestion
func NewRedisActivityRepository() ActivityService {
```

Reply:
Although there is not much in this struct, it's best to return a pointer from the constructor.

Reply:
1. This constructor encapsulates the underline redis implementation (via the interface) so I don't think that including "redis" in the name is necessary. 
2. I get an error when I try to return a pointer